wandb/sdk/artifacts/artifact.py

```diff
@@ -532,6 +532,27 @@ def type(self) -> str:
         """The artifact's type. Common types include `dataset` or `model`."""
         return self._type
 
+    @property
+    def url(self):
+        """
+        Constructs the URL of the artifact.
+
+        Returns:
+            str: The URL of the artifact.
+        """
```

Comment:
**(blocking):** Lets add a check to make sure that every part of the url we're returning is not `None. You can use python's `all([])` method here. Similar to how its done for runs: https://github.com/wandb/wandb/blob/e43471d9e090af801223afc5c0c639c978034eb8/wandb/sdk/wandb_settings.py#L885-L886

Reply:
Added to each of the new internal helper methods